pkg/publisher/webhookPublisher.go

```diff
@@ -80,50 +77,51 @@ func (p *WebhookPublisher) svc() {
 
 func (p *WebhookPublisher) makeHttpRequest(r *publishRequest) {
 	url := p.baseUrl + "/" + strings.TrimPrefix(r.target, "/")
-	p.logger.Info("making HTTP request", zap.String("url", url))
+
+	msg := "making HTTP request"
+	level := zap.ErrorLevel
+	fields := []zap.Field{zap.String("url", url), zap.String("type", "publish_request")}
+
+	// log once for this request
+	defer func() {
+		if ce := p.logger.Check(level, msg); ce != nil {
+			ce.Write(fields...)
+		}
+	}()
 
 	var buf bytes.Buffer
 	buf.WriteString(r.body)
 
 	// Create request
-	req, err := http.NewRequest("POST", url, &buf)
+	req, err := http.NewRequest(http.MethodPost, url, &buf)
 	if err != nil {
-		p.logger.Error("error creating request", zap.Error(err), zap.String("url", url))
+		fields = append(fields, zap.Error(err))
+		return
 	}
 	for k, v := range r.headers {
 		req.Header.Set(k, v)
 	}
-
 	// Make the request
 	resp, err := http.DefaultClient.Do(req)
-
-	// All done if the request succeeded with 200 OK.
-	if err == nil && resp.StatusCode == 200 {
-		resp.Body.Close()
-		return
-	}
-
-	// Log errors
+	var body []byte
```

Comment:
How about moving this line jsut right before `body, err = ioutil.ReadAll(resp.Body)`? 
```
} else {
    var body []byte
    body, err = ioutil.ReadAll(resp.Body)
    ...
```

Reply:
Have done as you suggest.